superset-frontend/src/components/AnchorLink/index.tsx

```diff
@@ -69,29 +53,35 @@ class AnchorLink extends React.PureComponent {
     }
   }
 
-  render() {
-    const { anchorLinkId, filters, showShortLinkButton, placement } =
-      this.props;
-    return (
-      <span className="anchor-link-container" id={anchorLinkId}>
-        {showShortLinkButton && (
-          <URLShortLinkButton
-            url={getDashboardUrl({
-              pathname: window.location.pathname,
-              filters,
-              hash: anchorLinkId,
-            })}
-            emailSubject={t('Superset chart')}
-            emailContent={t('Check out this chart in dashboard:')}
-            placement={placement}
-          />
-        )}
-      </span>
-    );
-  }
-}
+  useEffect(() => {
+    if (inFocus) {
+      scrollToView();
+    }
+  }, [inFocus])
 
-AnchorLink.propTypes = propTypes;
-AnchorLink.defaultProps = defaultProps;
+  useEffect(() => {
+    const hash = getLocationHash();
+    if (hash && anchorLinkId === hash) {
+      scrollToView();
+    }
+  });
```

Comment:
Should we pass an empty array as the second argument to avoid an accidental infinite loop just in case?

Reply:
since this is onMount, i think we do want an empty array

Reply:
Ok, will make the changes and update the PR shortly.